bindings/dart/lib/src/web/reranker/data_provider.dart

```diff
@@ -4,8 +4,20 @@ import 'package:xayn_ai_ffi_dart/src/common/reranker/data_provider.dart'
     as common show Asset, AssetType, baseAssets, SetupData;
 
 /// Returns a map of all assets required for initializing [`XaynAi`].
-Map<common.AssetType, common.Asset> getAssets() {
-  return common.baseAssets;
+Map<common.AssetType, common.Asset> getAssets(
+    {List<String> features = const []}) {
+  final redundantAssets = features.contains('webParallel')
+      ? [
+          common.AssetType.wasmSequentialModule,
+          common.AssetType.wasmSequentialScript,
+        ]
+      : [
+          common.AssetType.wasmParallelModule,
+          common.AssetType.wasmParallelScript,
+          common.AssetType.wasmParallelSnippet,
+        ];
```

Comment:
There was the idea to have a uniform naming/interface returned from `getAssets` independent of parallel/sequential.

I.e. for either of `wasmSequentialModule` or `wasmParallelModule` get mapped to `wasmModule`, same for `wasm...Script` => `wasmScript` and `wasm...Snippet` to `wasmSnippet`(1).

(Then you also don't need to pass the `features` to `SetupData`)

(1): It's not given in the sequential version but that might change, so basically "if there is a `wasmSnippet` is should be loaded.  (We might have one or multiple assets on sequential, too. In general we shouldn't consider what we might have if we can change it later, but if we do map anything from `xxxParallellYYY` to `xxxYYY` then for consistency why not do so for `Snippet` too :shrug: (or not if you don't want to).

Reply:
We can do this in another PR so we can close this.

Reply:
ok

Reply:
i agree, it's also not straight forward due to how and which variants of the `AssetType` enum are generated.